Add efibootmgr and unify EFI boot entry handling

Jkintalix opened the pull request:
- Added efibootmgr to core-packages-image.json package list
- Enhanced boot entry creation for both attended and Kickstart-based installs
- Updated boot label to Edge MicrovisorToolkit
- Improved EFI entry cleanup logic for existing boot entries

#### Merge Checklist  <!-- REQUIRED -->
**All** boxes should be checked before merging the PR
- [X] The changes in the PR have been built and tested
- [] cgmanifest file has been updated if required
- [X] Ready to merge

#### Description <!-- REQUIRED -->
Adds efibootmgr and improves EFI boot entry creation for both attended and Kickstart-based installs.
Updates boot label and cleanup logic to ensure consistent setup across install modes.

#### Any Newly Introduced Dependencies
None

#### How Has This Been Tested? <!-- REQUIRED -->
verified manually in all core platforms

yepmunchun commented on toolkit/tools/liveinstaller/liveinstaller.go:
```diff
@@ -180,15 +193,27 @@ func runBootEntryCreationCommand(installDetails installationDetails) (err error)
-		"|", "sed", "'s/* Azure Linux//g'", // Pruning for just the bootnum
-		"|", "sed", "'s/Boot*//g'", // Pruning for just the bootnum
-		"|", "xargs", "-t", "-i", "efibootmgr", "-b", "{}", "-B", // Calling efibootmgr --delete-bootnum (aka `-B`) on each pre-existing bootnum with an Azure Linux label
+	logger.Log.Info("Removing pre-existing 'Edge Microvisor Toolkit' boot targets from efibootmgr")
```
please update function name as well

Jkintalix replied:
Addressed

anujm1 commented on toolkit/imageconfigs/packagelists/grub2-mkconfig.json:
```diff
@@ -1,6 +1,7 @@
     "packages": [
-        "grub2"
+        "grub2",
+        "efibootmgr"
```
this is not related to `grub2-mkconfig` or `grub` .. why is it needed?

Jkintalix replied:
yes efibootmgr is not related to GRUB, so it has been moved to core-packages-images.json.

yepmunchun commented on toolkit/tools/liveinstaller/liveinstaller.go:
```diff
@@ -180,15 +193,27 @@ func runBootEntryCreationCommand(installDetails installationDetails) (err error)
+	testCmd := `efibootmgr | grep "Edge Microvisor Toolkit"`
+
+	// Actual deletion command
+	deleteCmd := `efibootmgr | grep "Edge Microvisor Toolkit" | awk '{print $1}' | sed 's/Boot//g' | sed 's/\*//g' | xargs -t -I {} efibootmgr -b {} -B`
```
is it possible to parse the output in Go, instead of using shell pipelines and command chaining?

Jkintalix replied:
Addressed

yepmunchun commented on toolkit/tools/liveinstaller/liveinstaller.go:
```diff
@@ -164,14 +164,27 @@ func runBootEntryCreationCommand(installDetails installationDetails) (err error)
+	if cfg.DefaultSystemConfig.IsKickStartBoot {
+		logger.Log.Info("Unattended installation: parsing Kickstart file to get disk info")
+
+		kickstartPartitionFile := "/tmp/part-include"
+		disks, _, err := configuration.ParseKickStartPartitionScheme(kickstartPartitionFile)
+		logger.PanicOnError(err, "Failed to parse partition schema")
+		// Save disk config settings
+		if len(disks) == 0 {
+			return fmt.Errorf("Kickstart parsed disk list is empty")
+		}
+		cfg.Disks = disks
```
this seems repeated, we have parsed and saved the Disks info in `imager.go:main()`, can you please check?

Jkintalix replied:
Initially, the unattended_config.json will not have disk information. Disk values are derived from the Kickstart partition file during the installation process and are not saved to the configuration file.
To handle this, we are implementing logic in the runbootcreation command to dynamically retrieve the disk information at runtime, which is then used to create the appropriate boot entry

Jkintalix commented:
@yepmunchun  and @anujm1 requesting for review.

yepmunchun commented on toolkit/tools/liveinstaller/liveinstaller.go:
```diff
@@ -164,32 +165,87 @@ func runBootEntryCreationCommand(installDetails installationDetails) (err error)
+	for _, id := range bootIDs {
+		delCmd := fmt.Sprintf("efibootmgr -b %s -B", id)
+		logger.Log.Infof("Deleting boot entry with command: %s", delCmd)
+		err := shell.ExecuteLive(squashErrors, "sh", "-c", delCmd)
```
I think we do not need `sh -c` in `shell.ExecuteLive`

Jkintalix replied:
Adressed

yepmunchun commented:
Consider not logging raw output at INFO level if it could be large or noisy, DEBUG level might be more appropriate.

Please cleanup logging:
If it is not needed, remove it
If it is low level info, move it to DEBUG level
Leave only informational type to INFO level

Jkintalix committed:
Add efibootmgr and unify EFI boot entry handling

- Added efibootmgr to grub2-mkconfig package list
- Enhanced boot entry creation for both attended and Kickstart-based installs
- Updated boot label to Edge MicrovisorToolkit
- Improved EFI entry cleanup logic for existing boot entries

Signed-off-by: kinatli jayanth <jayanthx.kintali@intel.com>
